### DISCUSSION TITLE
Chart/Graph Libraries that integrate with MDX

### DISCUSSION THREAD

**Discussion by Mochael** [timlrx/tailwind-nextjs-starter-blog#691] [Q&A]
Anybody use any libraries for adding pretty charts/graphs in your MDX files? I've been trying to embed some charts in my MDX files using various chart libraries (plotly, Nivo, Chartjs) with no luck

**Comment by Mochael** [comment]
I was eventually able to get a chart rendering after I:
- Moved the chart logic to be a child component of the component that I reference in the MDX
- Added 'use-client' to the parent component that I'm importing into the MDX file
- Passed in parent component into the MDXComponents in `components/MDXComponents.tsx`

**Reply by timlrx** [reply]
Glad that you managed to figure it out! I should add a section in the readme on rendering custom components in MDX. I think the two main things that might not be immediately apparent is the need to re-export the component with the `use-client` directive, unless it does not use any react hooks at all, and that it has to be default exported.

**Comment by timlrx** [comment]
Added a section in the readme for those that might be interested in implementing this: https://github.com/timlrx/tailwind-nextjs-starter-blog#how-can-i-add-a-custom-mdx-component
